pkg/controller/core/workload_controller.go

```diff
@@ -162,13 +162,13 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 				updated = true
 			case kueue.WorkloadEvictedByPodsReadyTimeout:
 				var requeueAfter time.Duration
-				if wl.Status.RequeueState != nil && wl.Status.RequeueState.RequeueAt != nil {
+				if workload.HasRequeueState(&wl) && wl.Status.RequeueState.RequeueAt != nil {
```

Comment:
I'm wondering what  the benefit of the nil-check wrapper is, can we remove it? 

It does not make code visually shorter as `wl.Status.RequeueState != nil` and `workload.HasRequeueState(&wl)` are the same length. 

If we introduce such a layer of abstraction I would also expect to have functions such as `workload.ClearRequeueState`. Otherwise it feels inconsistent - check by function, but direct assignment.

Reply:
That makes sense. Let's convert this PR to remove the `HasRequeueState` helper.